Client/Application/Migration.swift

```diff
@@ -26,8 +26,25 @@ class Migration {
     }
     
     static func moveDatabaseToApplicationDirectory() {
-        //Moves Coredata sqlite file from Documents dir to application support dir.
-        DataController.shared.migrateToNewPathIfNeeded()
+        if Preferences.Database.DocumentToSupportDirectoryMigration.completed.value {
+            // Migration has been done in some regard, so drop out.
+            return
+        }
+        
+        if Preferences.Database.DocumentToSupportDirectoryMigration.previousAttemptedVersion.value == AppInfo.appVersion {
+            // Migration has already been attempted for this version.
```

Comment:
We could convert some of these comments to debug logs

Reply:
Yes, true, I'll place some logs in. We need to massively cleanup our logs in general too, lol.